src/client/debugger/jupyter/debugControllers.ts

```diff
@@ -57,13 +57,16 @@ export class RunByLineController implements IDebuggingDelegate {
     public continue(): void {
         if (typeof this.lastPausedThreadId !== 'number') {
             traceVerbose(`No paused thread, can't do RBL`);
+            this.stop();
             return;
         }
 
         void this.debugAdapter.stepIn(this.lastPausedThreadId);
     }
 
     public stop(): void {
+        // When debugpy gets stuck, running a cell fixes it and allows us to start another debugging session
+        void this.kernel.executeHidden('pass');
```

Comment:
Isn't this a hack, i think we should get to the bottom of this.

Reply:
I'll file a bug in ipykernel, and mention it here, but I don't see what else we can do